webhookValidator_test.go

```diff
@@ -147,11 +146,19 @@ func TestCheckDuration(t *testing.T) {
 	for _, tc := range tcs {
 		t.Run(tc.desc, func(t *testing.T) {
 			assert := assert.New(t)
-			err := CheckDuration(tc.ttl)(tc.webhook)
-			assert.True(errors.Is(err, tc.expectedErr),
-				fmt.Errorf("error [%v] doesn't contain error [%v] in its err chain",
-					err, tc.expectedErr),
-			)
+			f, err := CheckDuration(tc.ttl)
```

Comment:
We should differentiate between this error and an error after calling `f()`.  If we expect `CheckDuration()` to return an error but it doesn't, we're not ready to call `f()` in that unit test and should just fail the test case at that point.

Reply:
I would structure it like this:
```go
f, err := CheckDuration(tc.ttl)
if tc.expectedInitialErr != nil {
  assert.True(/* error checking */)
  assert.Nil(f)
  return // this returns out of this single test case, not the larger test
}
require.NoError(err)
require.NotNil(f)
err = f(tc.webhook)
assert.Equal(/* error check stuff */)
```